posts/inside-rust/2023-09-29-polonius-update.md

```diff
@@ -0,0 +1,152 @@
+
+In the shorter term, we can refactor the borrow checker to separate out the higher-ranked processing from the ordinary processing. The goal would be to preprocess the outlives constraints in a kind of polonius ["leak check"](https://rustc-dev-guide.rust-lang.org/traits/hrtb.html#basic-matching-and-placeholder-leaks), where we can compute the higher-ranked errors. This can then be removed once the trait solver can solve these constraints.
+
+Current status: ⏳ members of the types team are starting to work on this task in the next few days.
+
+
+### 2. Location-insensitive loans in scope
+
+Out of the two key differences between polonius and the existing borrow check (regions as "sets of loans", and computing subtyping relations at each point in the CFG), this step is aimed at resolving the *first* difference, but not the second, so we call it the "location *in*sensitive loans in scope" (because subtyping is being done once, not per location): the idea can be described as "NLLs with the polonius model".
+
+Note that other aspects of the existing borrow checker are still flow-sensitive. 
+
+In this step, we will compute the set of live loans via outlives constraints only, instead of computing the CFG points where regions are live (which is then used to compute when loans go out of scope). We believe this is equivalent to the existing borrow check in terms of the errors that get reported.
+
+Importantly, this change paves the way for adding location sensitivity (sets of loans are also a better foundation for far-future improvements to the borrow checker such as safe internal references).
+
+Current status: ✅ we have completed prototypes, and have [an open PR](https://github.com/rust-lang/rust/pull/113218) to land this under a `-Z` flag, which should happen in the near future.
+
+### 3. Verify full test suite passes with location-insensitive polonius
+
+That PR does pass the full 15000+ tests in our suite, but we haven't yet checked on the crates published on crates.io with a crater run.
+
+Compared to our internal test suite, the vast majority of published crates are expected to build without errors. In that regard, it should be unlikely that issues would be found there, but it will be done regardless.
+
+Current status: ⏳ in-progress, should be finalized right when the PR is just ready to land.
```

Comment:
The wording here is a bit weird

Reply:
I've pushed a commit to reword this into the following, which should be clearer:

> Current status: ⏳ in-progress, the crater run itself will be done just before the PR lands.

Reply:
I probably would have removed the "just" here, but this is fine.

Reply:
I can do that.